internal/dag/gatewayapi_processor.go

```diff
@@ -112,6 +112,14 @@ func (p *GatewayAPIProcessor) Run(dag *DAG, source *KubernetesCache) {
 						}
 					}
 				}
+			} else {
```

Comment:
so this case now means a TLSRoute with no TLS field set defaults to terminate, though terminate without a secret doesn't really work? so can this just be an error now if you don't set TLS at all?

Reply:
/shrug The spec says that "Terminate" is the default.

Reply:
- https://github.com/kubernetes-sigs/gateway-api/blob/master/apis/v1alpha2/gateway_types.go#L321
- https://github.com/kubernetes-sigs/gateway-api/blob/master/apis/v1alpha2/gateway_types.go#L329-L330

Reply:
is this intended to be the code for if the mode is nil? rather than the whole TLS block is nil?

sounds like the listener should be rejected if the TLS field is not set: https://github.com/kubernetes-sigs/gateway-api/blob/17a98f6f5aaa1e7fa607c1fc974efeb915b3337c/apis/v1alpha2/gateway_types.go#L184-L197

Reply:
Essentially it is since the cert validation then fails on L118. Part of the problem with our unit tests is that we don't have the kubebuilder defaults applied. So we can create a test that doesn't specify the cert, but would get the mode applied by the api server and wouldn't pass the unit test.

Reply:
Ok I rereead the spec and at the top level it does say "p.Errorf("Listener.TLS is required when protocol is %q.", listener.Protocol)", so that make sense. I've updated.